src/query.rs

```diff
@@ -31,30 +41,39 @@ pub enum QueryResults {
     Rel(Vec<Vec<TypedValue>>),
 }
 
-pub enum QueryExecutionError {
-    ParseError(QueryParseError),
-    InvalidArgumentName(String),
-}
-
-impl From<QueryParseError> for QueryExecutionError {
-    fn from(err: QueryParseError) -> QueryExecutionError {
-        QueryExecutionError::ParseError(err)
-    }
-}
-
-pub type QueryExecutionResult = Result<QueryResults, QueryExecutionError>;
+pub type QueryExecutionResult = Result<QueryResults>;
 
 /// Take an EDN query string, a reference to a open SQLite connection, a Mentat DB, and an optional
 /// collection of input bindings (which should be keyed by `"?varname"`), and execute the query
 /// immediately, blocking the current thread.
 /// Returns a structure that corresponds to the kind of input query, populated with `TypedValue`
 /// instances.
+/// The caller is responsible for ensuring that the SQLite connection is in a transaction if
+/// isolation is required.
 #[allow(unused_variables)]
-pub fn q_once(sqlite: SQLiteConnection,
-              db: DB,
-              query: &str,
-              inputs: Option<HashMap<String, TypedValue>>) -> QueryExecutionResult {
+pub fn q_once<'sqlite, 'schema, 'query>
+(sqlite: &'sqlite rusqlite::Connection,
+ schema: &'schema Schema,
+ query: &'query str,
+ inputs: Option<HashMap<String, TypedValue>>) -> QueryExecutionResult {
```

Comment:
Yeah, it is. But there's no good indenting rules for long scope generic lists, which is probably why people use `'a, 'b, 'c`!

There are for types: `<T> (… foo: T) where T: Some<Long<Thing>>`, of course.